packages/mattermost-redux/src/utils/post_list.ts

```diff
@@ -323,142 +322,100 @@ export function makeGenerateCombinedPost(): (state: GlobalState, combinedId: str
-                    allUserIds.push(...ids);
+    userActivities.forEach((activity) => {
+        if (isUsersRelatedPost(activity.postType)) {
+            const {postType, actorId, userIds, usernames} = activity;
+            if (usernames && userIds) {
+                messageData.push({postType, userIds: [...usernames, ...userIds], actorId});
+                if (userIds.length > 0) {
+                    allUserIds.push(...userIds);
                 }
-
                 if (usernames.length > 0) {
                     allUsernames.push(...usernames);
                 }
                 allUserIds.push(actorId);
-            });
+            } else {
+                throw new Error('Invalid Post activity data');
+            }
         } else {
-            if (!Array.isArray(values)) {
+            const {postType, actorId} = activity;
+            const userIds = actorId;
+            if (actorId) {
+                messageData.push({postType, userIds});
+                allUserIds.push(userIds);
+            } else {
                 throw new Error('Invalid Post activity data');
             }
-            messageData.push({postType, userIds: values});
-            allUserIds.push(...values);
         }
     });
-
-    messageData.sort(comparePostTypes);
-
     function reduceUsers(acc: string[], curr: string) {
         if (!acc.includes(curr)) {
             acc.push(curr);
         }
         return acc;
     }
-
     return {
         allUserIds: allUserIds.reduce(reduceUsers, []),
         allUsernames: allUsernames.reduce(reduceUsers, []),
         messageData,
     };
 }
-
+function isUsersRelatedPost(postType: string) {
+    return (
+        postType === Posts.POST_TYPES.ADD_TO_TEAM ||
+        postType === Posts.POST_TYPES.ADD_TO_CHANNEL ||
+        postType === Posts.POST_TYPES.REMOVE_FROM_CHANNEL
+    );
+}
 export function combineUserActivitySystemPost(systemPosts: Post[] = []) {
     if (systemPosts.length === 0) {
         return null;
     }
-
-    const userActivities = systemPosts.reduce((acc: any, post: Post) => {
+    const userActivities: ActivityEntry[] = [];
+    let prevPost: any = {};
+    systemPosts.reverse().forEach((post: Post) => {
         const postType = post.type;
-        let userActivityProps = acc;
-        const combinedPostType = userActivityProps[postType as string];
-
-        if (
-            postType === Posts.POST_TYPES.ADD_TO_TEAM ||
-            postType === Posts.POST_TYPES.ADD_TO_CHANNEL ||
-            postType === Posts.POST_TYPES.REMOVE_FROM_CHANNEL
-        ) {
+        let actorId: string = post.user_id;
+        if (isUsersRelatedPost(postType)) {
+            // sets the propsUserId to an empty string, this allows the function to combine different "remove" messages with different actors.
+            if (postType === Posts.POST_TYPES.REMOVE_FROM_CHANNEL) {
+                actorId = '';
+            }
+            if (!post.user_id || !post.props) {
```

Comment:
Could you remove this error? I don't know exactly why the old code here liked to throw errors on bad input, but it was probably because it was written pre-TypeScript where it was a lot easier for us to accidentally pass bad input